model/host/host.go

```diff
@@ -1344,14 +1356,26 @@ func (h *Host) SetDNSName(ctx context.Context, dnsName string) error {
 		},
 		bson.M{
 			"$set": bson.M{
-				DNSKey: dnsName,
+				DNSKey:        params.Hostname,
+				ZoneKey:       params.Zone,
+				StartTimeKey:  params.LaunchTime,
+				PublicIPv4Key: params.PublicIPv4,
+				IPv4Key:       params.PrivateIPv4,
+				IPKey:         params.IPv6,
+				VolumesKey:    params.Volumes,
```

Comment:
I think in this case, we would only want to set the non-zero fields here, no? The original logic only set one field, and it had the guard above to make sure that the field was really populated, but if we have partial EC2 metadata, we definitely don't want to potentially overwrite what the cloud host ready job has already populated.